### PR TITLE
Bugfixes and syntax improvements

### PR DESCRIPTION
**Pull request by franramirez688**
General refactor in `bwproject.py`

### CODE REVIEW

File: bwdata.py [hunk 1]
```diff
@@ -72,7 +75,7 @@ def iter_mentions(self, name=None, startDate=None, max_pages=None, iter_by_page=
         Returns:
             A list of mentions.
         """
-        params = self._fill_params(name, startDate, kwargs)
+        params = self._fill_params(name, start_date, kwargs)
         params["pageSize"] = kwargs["pageSize"] if "pageSize" in kwargs else 5000
```
Comment (champ) [inline bwdata.py]: Let’s use `page_size` as `kwarg` for consistency, and use `get()` to retrieve its value and default value if not present.

File: bwdata.py [hunk 2]
```diff
@@ -135,303 +140,311 @@ def get_chart(self, name=None, startDate=None, y_axis=None, x_axis=None, breakdo
...
 
-        params = self._fill_params(name, startDate, kwargs)
+        params = self._fill_params(name, start_date, kwargs)
         if "dim1Args" in params:
             params["dim1Args"] = self._name_to_id(x_axis, params["dim1Args"])
         if "dim2Args" in params:
             params["dim2Args"] = self._name_to_id(breakdown_by, params["dim2Args"])
 
-        return self.project.get(endpoint="data/"+y_axis+"/"+x_axis+"/"+breakdown_by, params=params)
+        return self.project.get(endpoint="data/" + y_axis + "/" + x_axis + "/" + breakdown_by,
```
Comment (champ) [inline bwdata.py]: Change to use format to compose endpoint string

File: bwdata.py [hunk 3]
```diff
@@ -827,23 +855,27 @@ def _fill_params(self, name, startDate, data):
...
-            raise RuntimeError("No valid {} ids could be extracted".format(self.resource_type), name)
+            raise RuntimeError("No valid {} ids could be extracted".format(self.resource_type),
+                               name)
 
-        filled = {}
+        filled = dict()
         filled[self.resource_id_name] = id_list
-        filled["startDate"] = startDate
+        filled["startDate"] = start_date
         filled["endDate"] = data["endDate"] if "endDate" in data else (
```
Comment (champ) [inline bwdata.py]: Change to `end_date` in `data` for consistency. Use `get()` to get value or default value.

File: bwproject.py [hunk 4]
```diff
@@ -2,92 +2,95 @@
...
+            self._write_auth(token_path)
         elif username is not None:
-            self.username, self.token = self._read_auth(username, token_path)
+            self._read_auth(username, token_path)
         else:
-            raise KeyError("Must provide valid token, username and password, or username and path to token file")
+            raise KeyError("Must provide valid token, username and password,"
+                           " or username and path to token file")
+
+    def _update_by_test_auth(self, username, token):
```
Comment (champ) [inline bwproject.py:56]: Add docstring

**Review by champ — CHANGES_REQUESTED** [review]
@franramirez688  Reviewed

**Review by franramirez688 — COMMENTED** [review]
@champ review again

File: bwproject.py (hunk 4 above)
Comment (franramirez688) [reply bwproject.py:56]: It was already added

**Pull request merged**